subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryWalkerTest.groovy

```diff
@@ -118,6 +121,30 @@ class DirectoryWalkerTest extends AbstractDirectoryWalkerTest<DirectoryWalker> {
         }
     }
 
+    @Requires(TestPrecondition.SYMLINKS)
+    @Unroll
+    def "missing symbolic link causes an exception - walker: #walkerInstance.class.simpleName"() {
+        given:
+        def rootDir = tmpDir.createDir("root")
+        def dir = rootDir.createDir("a/b")
+        def link = rootDir.file("a/d")
+        link.createLink(dir)
+
+        when:
+        dir.deleteDir()
+        walkDirForPaths(walkerInstance, rootDir, new PatternSet())
+
+        then:
+        def e = thrown Exception
+        e.message.contains("Could not list contents of '${link.absolutePath}'.")
+
+        cleanup:
+        link.delete()
```

Comment:
Is this cleanup necessary?

Reply:
Most broken symlinks tests clean up after themselves, this test was just moved, I didn't want to change it:
https://github.com/gradle/gradle/blob/1f5fe429c16c7247b3267ba095f3f5b5b9d1298a/subprojects/files/src/testFixtures/groovy/org/gradle/api/internal/file/collections/AbstractDirectoryWalkerTest.groovy#L121

Would you like me to delete the cleanup in all symlink tests ?